server/channels/api4/channel_category_test.go

```diff
@@ -550,6 +551,791 @@ func TestUpdateCategoriesForTeamForUser(t *testing.T) {
+		require.NotNil(t, err)
+		require.Equal(t, http.StatusBadRequest, err.StatusCode)
+	})
+
+	t.Run("should handle empty channel list", func(t *testing.T) {
+		user, _ := setupUserForSubtest(t, th)
+
+		c := &Context{
+			App:        th.App,
+			AppContext: th.Context,
+			Logger:     th.App.Log(),
+		}
+
+		category := &model.SidebarCategoryWithChannels{
+			SidebarCategory: model.SidebarCategory{
+				UserId: user.Id,
+				TeamId: th.BasicTeam.Id,
+			},
+			Channels: []string{},
+		}
+
+		err := validateSidebarCategory(c, th.BasicTeam.Id, user.Id, category)
+		require.Nil(t, err)
+		require.Empty(t, category.Channels)
+	})
+}
+
+func TestValidateSidebarCategoryChannels(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	t.Run("should filter valid channels", func(t *testing.T) {
+		c := &Context{
+			App:        th.App,
+			AppContext: th.Context,
+			Logger:     th.App.Log(),
+		}
+
+		// Create test channels
+		channels := model.ChannelList{
+			th.BasicChannel,
+			th.BasicChannel2,
+		}
+
+		// Test with valid channel IDs
+		channelIds := []string{
+			th.BasicChannel.Id,
+			th.BasicChannel2.Id,
+		}
+
+		filtered := validateSidebarCategoryChannels(c, th.BasicUser.Id, channelIds, channels)
+		require.Len(t, filtered, 2)
+		require.ElementsMatch(t, channelIds, filtered)
+	})
+
+	t.Run("should filter out invalid channels", func(t *testing.T) {
+		c := &Context{
+			App:        th.App,
+			AppContext: th.Context,
+			Logger:     th.App.Log(),
+		}
+
+		channels := model.ChannelList{
+			th.BasicChannel,
+		}
+
+		channelIds := []string{
+			th.BasicChannel.Id,
+			"invalid_channel_id",
+		}
+
+		filtered := validateSidebarCategoryChannels(c, th.BasicUser.Id, channelIds, channels)
+		require.Len(t, filtered, 1)
+		require.Contains(t, filtered, th.BasicChannel.Id)
+		require.NotContains(t, filtered, "invalid_channel_id")
+	})
+
+	t.Run("should handle empty channel list", func(t *testing.T) {
+		c := &Context{
+			App:        th.App,
+			AppContext: th.Context,
+			Logger:     th.App.Log(),
+		}
+
+		channels := model.ChannelList{}
+		channelIds := []string{th.BasicChannel.Id}
+
+		filtered := validateSidebarCategoryChannels(c, th.BasicUser.Id, channelIds, channels)
+		require.Empty(t, filtered)
+	})
+
+	t.Run("should handle empty channelIds", func(t *testing.T) {
+		c := &Context{
+			App:        th.App,
+			AppContext: th.Context,
+			Logger:     th.App.Log(),
+		}
+
+		channels := model.ChannelList{
+			th.BasicChannel,
+			th.BasicChannel2,
+		}
+
+		filtered := validateSidebarCategoryChannels(c, th.BasicUser.Id, []string{}, channels)
+		require.Empty(t, filtered)
+	})
+
+	t.Run("should handle nil inputs", func(t *testing.T) {
+		c := &Context{
+			App:        th.App,
+			AppContext: th.Context,
+			Logger:     th.App.Log(),
+		}
+
+		filtered := validateSidebarCategoryChannels(c, th.BasicUser.Id, nil, nil)
+		require.Empty(t, filtered)
+	})
+
+	t.Run("should filter duplicate channel IDs", func(t *testing.T) {
```

Comment:
I think this test is named incorrectly since it makes it sound like the function should remove duplicates